ekncontent/tests/ekncontent/testDatadir.js

```diff
@@ -0,0 +1,14 @@
+const Eknc = imports.gi.EosKnowledgeContent;
+
+describe('Data dir finder', function () {
+    it('returns null if the domain was not found', function () {
+        let found_dir = Eknc.get_data_dir('this-knowledge-app-should-never-exist');
+        expect(found_dir).toBeNull();
+    });
+
+    // FIXME: no easy way to test these without a way to mock file URIs
+    // during tests.
+    it('returns a GFile pointing to a valid ekn data dir');
```

Comment:
Kinda thinking out loud here, but maybe a way to do this would be to define some method or environment variable...
`EKN_CONTENT_USE_TEST_URI`
or
`eknc_set_use_test_uri (TRUE)`
which would load everything with a `test://` uri, we could then do a `vfs.register_uri_scheme('test')` to serve whatever content we want.

Could play with that more. But since it involves more vfs stuff I'd rather wait till all that shakes out.

Alternately could pass a prefix along to `Eknc.get_data_dir` to use instead of the rootfs if set. And then write some directories to disk like the old tests did. But I am not a fan of either changing the exposed API for tests, or writing to disk during tests if avoidable.